sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -641,59 +615,174 @@ func withSuperpowered(superpowered bool) oci.SpecOpts {
+		{
+			Destination: "/sys/fs/cgroup",
+			Type:        "cgroup",
+			Source:      "cgroup",
+			Options:     []string{"ro", "nosuid", "noexec", "nodev"},
+		},
+	}
+
+	// The `current` dir was added for easier referencing in Dockerfiles and scripts.
+	// If a host container is also named `current`, only add a single `current` mount
+	// to the spec.
 	if containerID != "current" {
-		persistentMounts = append(persistentMounts, runtimespec.Mount{
+		mounts = append(mounts, runtimespec.Mount{
 			Options:     []string{"rbind", "rw"},
 			Destination: fmt.Sprintf("/.bottlerocket/%s/current", persistentDir),
 			Source:      fmt.Sprintf("/local/%s/%s", persistentDir, containerID),
 		})
 	}
-	return oci.Compose(oci.WithMounts(persistentMounts))
+
+	// Use withMounts to make sure all mounts have rprivate propagations
+	return withMounts(mounts)
+}
+
+// withPrivilegedMounts adds options to grant access to the host root filesystem
+func withPrivilegedMounts() oci.SpecOpts {
+	// Use withMounts to force rprivate when no propagation configurations
+	// are set
+	return withMounts([]runtimespec.Mount{
+		{
+			Options:     []string{"rbind", "ro"},
+			Destination: "/.bottlerocket/rootfs",
+			Source:      "/",
+			Type:        "bind",
+		},
+		{
+			Options:     []string{"rbind", "ro"},
+			Destination: "/lib/modules",
+			Source:      "/lib/modules",
+			Type:        "bind",
+		},
+		{
+			Options:     []string{"rbind", "rw"},
+			Destination: "/usr/src/kernels",
+			Source:      "/usr/src/kernels",
+			Type:        "bind",
+		},
+		{
+			Options:     []string{"rbind"},
+			Destination: "/sys/kernel/debug",
+			Source:      "/sys/kernel/debug",
+			Type:        "bind",
+		},
+		// Use shared propagations so mounts in this mount point propagate
+		// across the peer group
+		{
+			Options:     []string{"rbind", "rshared"},
+			Destination: "/.bottlerocket/rootfs/mnt",
+			Source:      "/mnt",
+			Type:        "bind",
+		},
+	})
+}
+
+// withMounts sets the mounts' propagations as rprivate only when the
+// mounts' options don't have propagations settings
+func withMounts(mounts []runtimespec.Mount) oci.SpecOpts {
+	finalMounts := []runtimespec.Mount{}
+
+	for _, mount := range mounts {
+		// Only set rprivate when no propagations are configured for
+		// the mount
+		if !hasPropagation(mount) {
+			// Update the local mount copy instead of the original
+			mount.Options = append(mount.Options, "rprivate")
+		}
+		finalMounts = append(finalMounts, mount)
+	}
+
+	return oci.WithMounts(finalMounts)
+}
+
+// hasPropagation checks if the mount has propagation options
+func hasPropagation(mount runtimespec.Mount) bool {
+	var hasPropagation = false
+	// Propagations can be shared, rshared, private, rprivate, slave, rslave
+	re := regexp.MustCompile(`r?(shared|private|slave)`)
+
+	for _, option := range mount.Options {
+		hasPropagation = re.FindString(option) != ""
+
+		if hasPropagation {
+			break
+		}
+	}
+
+	return hasPropagation
```

Comment:
I'd prefer to avoid the use of a regular expression here.  There are only six values we need to test for.

```suggestion
	// Propagations can be shared, rshared, private, rprivate, slave, rslave
	for _, option := range mount.Options {
		switch option {
		case "shared", "rshared", "private", "rprivate", "slave", "rslave":
			return true
		}
	}

	return false
```